website/docs/spec/index.md

```diff
@@ -179,6 +183,30 @@ The message encoding and schema must match that of the Channel record correspond
 | 8     | publish_time | Timestamp | Time at which the message was published. If not available, must be set to the log time.                         |
 | N     | data         | Bytes     | Message data, to be decoded according to the schema of the channel.                                             |
 
+### Secondary Index Key (op=0x10)
+
+A Secondary Index Key record defines a secondary timestamp index that will be used in this file.
+Secondary Indexes can be used to quickly look up messages by timestamps other than `log_time`.
+The `name` field identifies the timestamp key that messages will be indexed by. The [registry](./registry.md#secondary-index-keys) lists well-known secondary index key names.
+
+A Secondary Index Key record must appear before any [Secondary Message Index](#secondary-message-index-op0x11) records
+in the data section with this `secondary_index_id`.
+
+Secondary Index Key records in the Data section must also appear in the Summary section, before
+any [Secondary Chunk Index](#secondary-chunk-index-op0x12) records with this `secondary_index_id`.
+
+| Bytes | Name               | Type   | Description                                                       |
+| ----- | ------------------ | ------ | ----------------------------------------------------------------- |
+| 2     | secondary_index_id | uint16 | A unique identifier for this secondary index within the file.     |
+| 4 + N | name               | string | A name that describes the key, eg. `publish_time`, `header.stamp` |
+
+> Why do Secondary Index Key records appear in the Data section?
+> When reading using an index, the Secondary Index Key would be read out of the Summary section
+> before reading into the Data section. This means that the Secondary Index Key in the Data section
+> is not normally used. However, if a MCAP is truncated and the summary section is lost, having the
+> Secondary Index Key appear before any Secondary Message Index records allows the MCAP to be fully
+> recovered.
```

Comment:
Why is it necessary to have this record type, vs the parser just inferring from the existence of the other two record types that the index is in use? I can see dropping this would require moving the "name" field into the secondary chunk index record but that doesn't seem like the biggest thing we stick in those records anyway.

Today the parser knows a file is indexed via the presence of the index records - we don't need a third record type for that right?

Reply:
forget the "metadata" part - I think it would be better implemented with a "chunk info" record as described in another comment.